test/Microsoft.DocAsCode.Tests.Common/TestLoggerListener.cs

```diff
@@ -15,22 +15,29 @@ public class TestLoggerListener : ILoggerListener
 
         public LogLevel LogLevelThreshold { get; set; }
 
-        public TestLoggerListener(string phase = null)
-        {
-            Phase = phase;
-        }
+        public bool EnablePhaseEndWith { get; set; }
 
-        public void Dispose()
+        public TestLoggerListener()
         {
         }
 
-        public void Flush()
+        public TestLoggerListener(string phase, LogLevel logLevelThreshold = LogLevel.Warning, bool enablePhaseEndWith = false)
         {
+            Phase = phase;
+            LogLevelThreshold = logLevelThreshold;
+            EnablePhaseEndWith = enablePhaseEndWith;
         }
 
         public void WriteLine(ILogItem item)
         {
-            if (Phase == null || item.Phase == Phase)
+            if (item == null || item.LogLevel < LogLevelThreshold)
+            {
+                return;
+            }
+            if (Phase == null ||
+                item.Phase == Phase ||
+                !EnablePhaseEndWith && item.Phase.StartsWith(Phase) ||
```

Comment:
Or a Match delegate?

---
In reply to: [119277973](https://github.com/dotnet/docfx/pull/1712#discussion_r119277973) [](ancestors = 119277973)